tasks/partial_bundle.js

```diff
@@ -7,41 +8,78 @@ var common = require('./util/common');
 var _bundle = require('./util/browserify_wrapper');
 
 var header = constants.licenseDist + '\n';
+var allTransforms = constants.allTransforms;
 var allTraces = constants.allTraces;
 var mainIndex = constants.mainIndex;
 
-var argv = process.argv;
-
-if(argv.length > 2) {
-    // command line
-
-    var traceList = ['scatter']; // added by default
-    var name;
-    for(var i = 2; i < argv.length; i++) {
-        var a = argv[i];
-
+function createList(outList, inList, allList, type) {
+    for(var i = 0; i < inList.length; i++) {
+        var t = inList[i];
         if(
-            allTraces.indexOf(a) !== -1 && // requested
-            traceList.indexOf(a) === -1    // not added before
+            outList.indexOf(t) === -1 // not added before
         ) {
-            traceList.push(a);
+            if(allList.indexOf(t) === -1) {
+                console.error(t, 'is not a valid ' + type + '!', 'Valid ' + type + 's are:', allList);
+            } else {
+                outList.push(t);
+            }
         }
-        if(a.indexOf('--name=') === 0) name = a.replace('--name=', '');
     }
-    if(!name) name = 'custom';
-    traceList = traceList.sort();
+
+    return outList.sort();
+}
+
+function isFalse(a) {
+    return (
+        a === 'none' ||
+        a === 'false'
+    );
+}
+
+function inputBoolean(a, dflt) {
+    return !a ? dflt : !isFalse(a);
+}
+
+function inputArray(a, dflt) {
+    dflt = dflt.slice();
+
+    return (
+        isFalse(a) ? [] :
+            !a || a === 'all' ? dflt :
+                a.split(',')
+    );
+}
+
+if(process.argv.length > 2) {
+    // command line
+
+    var args = minimist(process.argv.slice(2), {});
+
+    // parse arguments
+    var unminified = inputBoolean(args.unminified, false);
+    var out = args.out ? args.out : 'custom';
+    var traces = inputArray(args.traces, allTraces);
+    var transforms = inputArray(args.transforms, allTransforms);
 
     var opts = {
-        traceList: traceList,
-        name: name,
+        traceList: createList(['scatter'], traces, allTraces, 'trace'),
+        transformList: createList([], transforms, allTransforms, 'transform'),
 
-        index: path.join(constants.pathToBuild, 'index-' + name + '.js'),
-        dist: path.join(constants.pathToDist, 'plotly-' + name + '.js'),
-        distMin: path.join(constants.pathToDist, 'plotly-' + name + '.min.js')
+        name: out,
+        index: path.join(constants.pathToLib, 'index-' + out + '.js')
     };
 
+    if(unminified) {
+        opts.dist = path.join(constants.pathToDist, 'plotly-' + out + '.js');
+    } else {
+        opts.distMin = path.join(constants.pathToDist, 'plotly-' + out + '.min.js');
```

Comment:
To avoid name conflicts (let's say one uses e.g. `basic` in output name` with different traces), wondering instead of `plotly-*` shouldn't we use `plotly_*` here?

Reply:
On the other hand folks making custom bundles will likely be used to our partial bundles already, so may be tripped up by the switch from `-` to `_`. Let's keep `-`, these bundles won't be published anyway.